spring-integration-core/src/main/java/org/springframework/integration/aot/CoreRuntimeHintsRegistrar.java

```diff
@@ -0,0 +1,157 @@
+import org.springframework.context.annotation.Bean;
+import org.springframework.core.DecoratingProxy;
+import org.springframework.integration.context.IntegrationContextUtils;
+import org.springframework.integration.core.GenericSelector;
+import org.springframework.integration.core.Pausable;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.gateway.MethodArgsHolder;
+import org.springframework.integration.gateway.RequestReplyExchanger;
+import org.springframework.integration.handler.AbstractReplyProducingMessageHandler;
+import org.springframework.integration.handler.DelayHandler;
+import org.springframework.integration.handler.GenericHandler;
+import org.springframework.integration.history.MessageHistory;
+import org.springframework.integration.json.JsonPathUtils;
+import org.springframework.integration.message.AdviceMessage;
+import org.springframework.integration.routingslip.ExpressionEvaluatingRoutingSlipRouteStrategy;
+import org.springframework.integration.store.MessageGroupMetadata;
+import org.springframework.integration.store.MessageHolder;
+import org.springframework.integration.store.MessageMetadata;
+import org.springframework.integration.support.MutableMessage;
+import org.springframework.integration.support.MutableMessageHeaders;
+import org.springframework.integration.transformer.GenericTransformer;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.messaging.support.ErrorMessage;
+import org.springframework.messaging.support.GenericMessage;
+
+/**
+ * {@link RuntimeHintsRegistrar} for Spring Integration core module.
+ *
+ * @author Artem Bilan
+ *
+ * @since 6.0
+ */
+class CoreRuntimeHintsRegistrar implements RuntimeHintsRegistrar {
+
+	@Override
+	public void registerHints(RuntimeHints hints, ClassLoader classLoader) {
+		ReflectionHints reflectionHints = hints.reflection();
+		Stream.of(
+						GenericSelector.class,
+						GenericTransformer.class,
+						GenericHandler.class,
+						Function.class,
+						Supplier.class,
+						BeanExpressionContext.class,
+						IntegrationContextUtils.class,
+						MethodArgsHolder.class,
+						AbstractReplyProducingMessageHandler.RequestHandler.class,
+						ExpressionEvaluatingRoutingSlipRouteStrategy.RequestAndReply.class,
+						Pausable.class,
+						SmartLifecycle.class)
+				.forEach(type ->
+						reflectionHints.registerType(type,
+								builder -> builder.withMembers(MemberCategory.INVOKE_PUBLIC_METHODS)));
+
+		reflectionHints.registerType(JsonPathUtils.class,
+				builder ->
+						builder.onReachableType(TypeReference.of("com.jayway.jsonpath.JsonPath"))
+								.withMembers(MemberCategory.INVOKE_PUBLIC_METHODS));
+
+		// For #xpath() SpEL function
+		reflectionHints.registerTypeIfPresent(classLoader, "org.springframework.integration.xml.xpath.XPathUtils",
+				builder -> builder.withMembers(MemberCategory.INVOKE_PUBLIC_METHODS));
+
+		Stream.of(
```

Comment:
It depends if those Kotlin types are used for Spring Integration specific features, in that case could make sense to keep them here.

@artembilan Could you share more context why those Kotlin hints are needed and where is the related code in Spring Integration?

Reply:
The related code is in the https://github.com/spring-projects/spring-integration/blob/main/spring-integration-core/src/main/java/org/springframework/integration/util/ClassUtils.java.

I need this info at runtime to determine if I deal with this or that lambda representation.
Having Kotlin as optional dependency I don't see other way unless I calculate it through the reflection and use later on respectively.

Reply:
Ok I think that's ok since those signals are Spring Integration specific.

Not sure if that make sense or not, but notice you could refine the implementation with boolean static flags to allow build time code removal when Kotlin is not in the classpath, see https://github.com/spring-projects/spring-framework/commit/f8508c805f2d24b1bf558d3becfb8ca8626d4d16. That that's only worth to if the code conditionally executed by those functions are significant enough.

Side question : are `isKotlinFaction0` and `isKotlinFaction1` typos (I would expect `isKotlinFuction0` and `isKotlinFuction1`)?

Reply:
Doh! I will fix those typos with deprecation in the previous version, removal in the current `6.0`.

Yes, I will consider to make those class variables conditional based on proposed `KotlinDetector`.

Thank you!

Reply:
So, I added `if (KotlinDetector.isKotlinPresent()) {` before trying to load those Kotlin classes I need in other places where I call methods via reflection.
Another complexity with this that Kotlin is optional dep, so I cannot call its methods directly unlike `java.util.function` classes...